nltk/util.py

```diff
@@ -505,6 +518,14 @@ def skipgrams(sequence, n, k):
     :type  k: int
     :rtype: iter(tuple)
     """
+    
+    # Pads the sequence as desired by **kwargs.
+    if 'pad_left' in kwargs or 'pad_right' in kwargs:
+        sequence = pad_sequence(sequence, n, **kwargs)
+    
+    # Note when iterating through the ngrams, the pad_right here is not
+    # the **kwargs padding, it's for the algorithm to detect None on the 
```

Comment:
Not due to your code, but `None` shouldn't be used as a sentinel (the word you're looking for here) when it can be avoided. `sentinel = object()` is more robust